index.js

```diff
@@ -42,19 +40,49 @@ const register = function(server, options) {
 
   server.decorate('request', 'timingEnd', function(key) {
     const timing = this.plugins['hapi-timing'][key];
-    timing.end = new Date();
-    timing.elapsed = timing.end - timing.start;
+    this.plugins['hapi-timing'][key].end = new Date();
+    this.plugins['hapi-timing'][key].elapsed = timing.end - timing.start;
   });
 
-  server.events.on('response', (request) => {
-    // check the tail response times and notify if needed:
-    const responseTime = request.info.responded - request.info.received;
-    const plugin = request.route.settings.plugins['hapi-timing'];
-    const threshold = (plugin && plugin.threshold) ? plugin.threshold : options.threshold;
-    if (responseTime > threshold) {
-      requestTimeoutExpired(responseTime, threshold, request);
-    }
-  });
+  if (options.requestLifecycle) {
+    const eventNames = ['onRequest', 'onPreAuth', 'onPostAuth', 'onPreHandler', 'onPostHandler'];
+    eventNames.forEach((eventName, i) => {
+      server.ext(eventName, (request, h) => {
+        if (i !== 0) {
+          // call timingEnd on previous event:
+          request.timingEnd(eventNames[i - 1]);
+        }
+        request.timingStart(eventName);
+        return h.continue;
+      });
+    });
+    server.events.on('response', (request) => {
+      request.timingEnd('onPostHandler');
+      // log the tardiness:
+      const responseTime = request.info.responded - request.info.received;
+      server.log(tags, {
```

Comment:
why do we need this twice?

Reply:
should be able to combine this with below. requestLifecycle should be just included with the log when the threshold is hit